tiles no sound build fails with `use of undeclared identifier` error

### Describe the bug

both gcc-14 and clang-18 linux tiles no sound builds fail with:
`Error: src/sdltiles.cpp:3726:9: error: use of undeclared identifier 'sound_init_success'` or `src/sdltiles.cpp:3726:9: error: ‘sound_init_success’ was not declared in this scope`
Probably caused by #76782 . [Clang 12, Ubuntu, Tiles, ASan](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/11146352718/job/30978148174?pr=76782) failed with same error


### Attach save file

N/A

### Steps to reproduce

make -j4 TILES=1 SOUND=0

### Expected behavior

no fatal errors

### Screenshots

_No response_

### Versions and configuration

linux x64
https://github.com/CleverRaven/Cataclysm-DDA/commit/f61d1cdb6a603d78410efb163e75593aababa1d6
clang version 18.1.8
gcc version 14.2.1

### Additional context

it is declared inside #if defined(SDL_SOUND) block. I guess wrapping this one use as well is a reasonable fix for this fix this. Declaring it outside of that block does not make sense
```patch
diff --git a/src/sdltiles.cpp b/src/sdltiles.cpp
index bea483bd98..6c42800984 100644
--- a/src/sdltiles.cpp
+++ b/src/sdltiles.cpp
@@ -3722,10 +3722,12 @@ void catacurses::init_interface()
     color_loader<SDL_Color>().load( windowsPalette );
     init_colors();
 
+#if defined(SDL_SOUND)
     // initialize sound set
     if( sound_init_success ) {
         load_soundset();
     }
+#endif
 
     font = std::make_unique<FontFallbackList>( renderer, format, fl.fontwidth, fl.fontheight,
             windowsPalette, fl.typeface, fl.fontsize, fl.fontblending );
```

--- a/src/sdltiles.cpp
+++ b/src/sdltiles.cpp
@@ -3722,10 +3722,12 @@ void catacurses::init_interface()
     color_loader<SDL_Color>().load( windowsPalette );
     init_colors();
 
+#if defined(SDL_SOUND)
     // initialize sound set
     if( sound_init_success ) {
         load_soundset();
     }
+#endif // SOUND
 
     font = std::make_unique<FontFallbackList>( renderer, format, fl.fontwidth, fl.fontheight,
             windowsPalette, fl.typeface, fl.fontsize, fl.fontblending );

